src/Validation/Indisposable.php

```diff
@@ -2,52 +2,120 @@
 
 namespace Propaganistas\LaravelDisposableEmail\Validation;
 
-use Illuminate\Support\Str;
+use Illuminate\Support\Facades\Cache as FrameworkCache;
+use Illuminate\Support\Facades\Log;
+use Exception;
+
+class Indisposable {
 
-class Indisposable
-{
     /**
-     * The array of disposable domains.
+     * The remote JSON source URI.
      *
-     * @var array
+     * @var string
+     */
+    protected $sourceUrl = 'https://rawgit.com/andreis/disposable-email-domains/master/domains.json';
+
+    /**
+     * Framework cache key.
+     *
+     * @var string
      */
-    protected static $domains = [];
+    protected $cacheKey = 'laravel-disposable-email.cache';
 
     /**
-     * Disposable constructor.
+     * Array of disposable email domains.
+     *
+     * @var array
      */
-    public function __construct()
-    {
-        if (empty(static::$domains)) {
-            $path = base_path('storage/framework/disposable_domains.json');
+    protected $domains = [];
 
-            // If the list hasn't been fetched yet, fall back to defaults.
-            // This ensures developers are up and running right away.
-            static::$domains = json_decode(file_get_contents(
-                file_exists($path) ? $path : (__DIR__ . '/../../domains.json')
-            ), true);
+    /**
+     * Indisposable constructor.
+     */
+    public function __construct() {
+        try {
+            $this->domains = $this->remoteDomains();
+        } catch (Exception $exception) {
+            Log::warning($exception->getMessage());
```

Comment:
No need to implement logging in a package. It's up to the developer to decide if logging is desired in the ExceptionHandleR.

Reply:
Got it! Thanks for the explanation. 👍